.github/workflows/validate_dataset.yml

```diff
@@ -0,0 +1,68 @@
+name: Validate Datasets
+
+# on .json PR to /data/datasets folder
+on: 
+  push:
+    secrets:
+      stac_ingestor_url:
+        required: true
+      client_id:
+        required: true
+      client_secret:
+        required: true
+      cognito_domain:
+        required: true
+      scope:
+        required: true
+    paths:
+      - 'data/datasets/**'
+
+# send pushed file as body to (API)/datasets/validate using secret credentials
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Filter for dataset files
+        uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          base: 'main'
+          list-files: shell
+          filters: |
+            dataset:
+              - 'data/datasets/*.json'
+      - name: Validate datasets using API
+        if: steps.filter.outputs.dataset == 'true'
+        env:
+          STAC_INGESTOR_URL: ${{ secrets.STAC_INGESTOR_URL }}
+        shell: bash
+        # LIMITATION - only reports errors in first CURL that fails
+        run: |
+          curl -X POST \
+            -H "Content-Type: application/x-www-form-urlencoded" \
+            -d "grant_type=client_credentials&scope=${{ secrets.scope }}" \
+            -u "${{ secrets.client_id }}:${{ secrets.client_secret }}" \
+            "${{ secrets.cognito_domain }}/oauth2/token" > token.txt
+          error_flag=0
+          for file in ${{ steps.filter.outputs.dataset_files }}
```

Comment:
Looking at the `ingestor`'s validate logic, I'm wondering whether it might be to o expensive to run for all datasets at the time of CI. Perhaps this requires a thinner, less robust validation that samples the data rather than poking every raster that may be in a bucket

Reply:
I think this is just a confusing quirk of `path-filters` - the `dataset_files` here represents the list of files that get picked up by the filter, so it's only the `json` inputs.